lib/rucio/api/rule.py

```diff
@@ -13,17 +13,30 @@
   - Patrick Austin, <patrick.austin@stfc.ac.uk>, 2020
   - Eli Chadwick, <eli.chadwick@stfc.ac.uk>, 2020
   - Ian Johnson, <ian.johnson@stfc.ac.uk>, 2021
+  - Radu Carpa <radu.carpa@cern.ch>, 2021
 
   PY3K COMPATIBLE
 '''
+from datetime import datetime
 
 from rucio.api.permission import has_permission
-from rucio.common.exception import AccessDenied
+from rucio.common.config import config_get_bool
+from rucio.common.exception import AccessDenied, InvalidObject
 from rucio.common.schema import validate_schema
 from rucio.common.types import InternalAccount, InternalScope
 from rucio.common.utils import api_update_return_dict
 from rucio.core import rule
-from rucio.common.config import config_get_bool
+
+
+def __parse_datetime(time_str):
+    if not time_str:
+        return None
+    try:
+        if hasattr(datetime, 'fromisoformat'):
+            return datetime.fromisoformat(time_str)
+        return datetime.strptime(time_str, '%Y-%m-%dT%H:%M:%S')
+    except ValueError as error:  # NOQA, pylint: disable=W0612
+        raise InvalidObject("Problem parsing time stamp: %(error)s" % locals())
```

Comment:
just wondering if repeating the correct time stamp format should also be added here?

something like      `expected format is YYYY-MM-DDTHH:MM:S`  ?

Reply:
Also careful with UTC here. (All timestamps in the DB should be in UTC)
For most other commands where we use time/duration, we always used basically durations. e.g. for rules `--lifetime` instead of `--expires-at` etc. We could do something similar here with a `--delay-injection <seconds>`

Reply:
Done. I initially tried to remove the `asynchronous` parameter from all core functions, but it became quickly very counter-intuitive to have to set `delay_injection=0` to mean "asynchronous". As a result, I keep both parameters, but ensure that setting delay_injection implies asynchronous